jujugui/static/gui/src/test/test_charmstore_apiv4.js

```diff
@@ -55,50 +55,17 @@ describe('Charmstore API v4', function() {
 
     it('calls the requestHandler to make a GET request', function() {
       var getRequest = utils.makeStubMethod(
-          charmstore.requestHandler, 'sendGetRequest');
-      var requestHandler = utils.makeStubMethod(charmstore, '_requestHandler');
+          charmstore.bakery, 'sendGetRequest');
       this._cleanups.concat([
-        getRequest.reset,
-        requestHandler.reset
+        getRequest.reset
       ]);
       charmstore._makeRequest('path', 'success', 'failure');
       assert.equal(getRequest.callCount(), 1);
       var getArgs = getRequest.lastArguments();
       assert.equal(getArgs[0], 'path');
-      assert.strictEqual(getArgs[1], null);
-      assert.strictEqual(getArgs[2], null);
-      assert.strictEqual(getArgs[3], null);
-      assert.strictEqual(getArgs[4], null);
-      // Make sure that the request handler is called with the callbacks.
-      getArgs[5]();
-      assert.equal(requestHandler.callCount(), 1);
-      assert.deepEqual(requestHandler.lastArguments(), [
-        'success', 'failure']);
-    });
-  });
-
-  describe('_requestHandler', function() {
-    var success, failure;
-
-    beforeEach(function() {
-      success = utils.makeStubFunction();
-      failure = utils.makeStubFunction();
-    });
-
-    it('calls the failure callback if status > 400', function() {
-      charmstore._requestHandler(success, failure, {
-        target: { status: 404 }
-      });
-      assert.equal(success.callCount(), 0);
-      assert.equal(failure.callCount(), 1);
-    });
-
-    it('calls the success callback if status < 400', function() {
-      charmstore._requestHandler(success, failure, {
-        target: { status: 200 }
-      });
-      assert.equal(success.callCount(), 1);
-      assert.equal(failure.callCount(), 0);
+      assert.strictEqual(getArgs[1], charmstore.charmstoreURL + 'v4/set-auth-cookie');
```

Comment:
Lets chat about this on Monday as I'm confused why simply needing a macaroon means that every call goes to the same endpoint instead of proper rest requests being rejected, at which point you go and refresh your macaroon.